packages/edit-site/src/components/list/added-by.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import { __experimentalHStack as HStack, Icon } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import {
+	layout as themeIcon,
+	plugins as pluginIcon,
+	commentAuthorAvatar as customizedByUserIcon,
+} from '@wordpress/icons';
+
+const TEMPLATE_POST_TYPE_NAMES = [ 'wp_template', 'wp_template_part' ];
+const NON_PLUGIN_SOURCES = [ 'theme', 'custom' ];
+
+function AddedByTheme( { slug } ) {
+	const theme = useSelect(
+		( select ) => select( coreStore ).getTheme( slug ),
+		[ slug ]
+	);
+
+	return (
+		<HStack alignment="left">
+			<div className="edit-site-list-added-by__icon">
+				<Icon icon={ themeIcon } />
+			</div>
+			<span>{ theme?.name?.rendered }</span>
+		</HStack>
+	);
+}
+
+function AddedByPlugin( { slug } ) {
+	const plugin = useSelect(
+		( select ) => select( coreStore ).getPlugin( slug ),
+		[ slug ]
+	);
+
+	return (
+		<HStack alignment="left">
+			<div className="edit-site-list-added-by__icon">
+				<Icon icon={ pluginIcon } />
+			</div>
+			<span>{ plugin?.name }</span>
+		</HStack>
+	);
+}
+
+function AddedByAuthor( { id } ) {
+	const user = useSelect( ( select ) => select( coreStore ).getUser( id ), [
+		id,
+	] );
+
+	return (
+		<HStack alignment="left">
+			<img
+				className="edit-site-list-added-by__avatar"
+				alt=""
+				src={ user?.avatar_urls[ 48 ] }
+			/>
+			<span>{ user?.nickname }</span>
+		</HStack>
+	);
+}
+
+export default function AddedBy( { templateType, template } ) {
+	if ( ! template ) {
+		return;
+	}
+
+	if ( TEMPLATE_POST_TYPE_NAMES.includes( templateType ) ) {
+		if ( template.has_theme_file && template.source === 'theme' ) {
+			return <AddedByTheme slug={ template.theme } />;
+		}
+
+		if ( ! NON_PLUGIN_SOURCES.includes( template.source ) ) {
+			return <AddedByPlugin slug={ template.source } />;
+		}
+
+		// Fallback for any custom template already created without an assigned
+		// author.
+		if (
+			! template.has_theme_file &&
+			template.source === 'custom' &&
+			! template.author
+		) {
+			return (
+				<HStack alignment="left">
+					<div className="edit-site-list-added-by__icon">
+						<Icon icon={ customizedByUserIcon } />
+					</div>
+					<span>Customized by user</span>
+				</HStack>
+			);
+		}
```

Comment:
Currently this is what displays for any templates that were customized but don't have an author.

I think even if we add author support to templates and template parts, there will still be some that have been created by Gutenberg plugin users up until now that won't have an author, so we need to handle this case.

@jameskoster Maybe the best option in this situation would be to show the theme icon and name, but still with the blue customized dot?

Reply:
> Currently this is what displays for any templates that were customized but don't have an author:

The important thing to communicate here is that the template was created from scratch. In lieu of a specific author, perhaps we could just say it was added by the site, and use the site logo as the graphic?

<img width="414" alt="Screenshot 2021-11-23 at 10 22 26" src="https://user-images.githubusercontent.com/846565/143007584-f197abb5-3b41-41b0-9915-69822757aaf7.png">

Reply:
Managed to make that work 👍 
<img width="721" alt="Screenshot 2021-11-25 at 6 43 56 pm" src="https://user-images.githubusercontent.com/677833/143427247-5c0c5357-88c7-4047-8087-0ff2759268e1.png">